pkg/providers/imagefamily/bootstrap/cse_cmd_nbcontract.sh.gtpl

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash
+
+set -o allexport # export all variables to subshells
+echo '#EOF' >> /opt/azure/manifest.json # wait_for_file looks for this
+mkdir -p /var/log/azure/Microsoft.Azure.Extensions.CustomScript/events # expected, but not created w/o CSE
+
+echo $(date),$(hostname) > /var/log/azure/cluster-provision-cse-output.log;
+for i in $(seq 1 1200); do
+grep -Fq "EOF" /opt/azure/containers/provision.sh && break;
+if [ $i -eq 1200 ]; then exit 100; else sleep 1; fi;
+done;
+{{if getIsAksCustomCloud .CustomCloudConfig}}
+for i in $(seq 1 1200); do
+grep -Fq "EOF" {{getInitAKSCustomCloudFilepath}} && break;
+if [ $i -eq 1200 ]; then exit 100; else sleep 1; fi;
+done;
+REPO_DEPOT_ENDPOINT="{{.CustomCloudConfig.GetRepoDepotEndpoint}}"
+{{getInitAKSCustomCloudFilepath}} >> /var/log/azure/cluster-provision.log 2>&1;
+{{end}}
+ADMINUSER={{getLinuxAdminUsername .GetLinuxAdminUsername}}
+MOBY_VERSION=
+TENANT_ID={{.AuthConfig.GetTenantId}}
+KUBERNETES_VERSION={{.GetKubernetesVersion}}
+KUBE_BINARY_URL={{.KubeBinaryConfig.GetKubeBinaryUrl}}
+CUSTOM_KUBE_BINARY_URL={{.KubeBinaryConfig.GetCustomKubeBinaryUrl}}
+PRIVATE_KUBE_BINARY_URL="{{.KubeBinaryConfig.GetPrivateKubeBinaryUrl}}"
+KUBEPROXY_URL={{.GetKubeProxyUrl}}
+APISERVER_PUBLIC_KEY={{.ApiServerConfig.GetApiServerPublicKey}}
+SUBSCRIPTION_ID={{.AuthConfig.GetSubscriptionId}}
+RESOURCE_GROUP={{.ClusterConfig.GetResourceGroup}}
+LOCATION={{.ClusterConfig.GetLocation}}
+VM_TYPE={{getStringFromVMType .ClusterConfig.GetVmType}}
+SUBNET={{.ClusterConfig.GetClusterNetworkConfig.GetSubnet}}
+NETWORK_SECURITY_GROUP={{.ClusterConfig.GetClusterNetworkConfig.GetSecurityGroupName}}
+VIRTUAL_NETWORK={{.ClusterConfig.GetClusterNetworkConfig.GetVnetName}}
+VIRTUAL_NETWORK_RESOURCE_GROUP={{.ClusterConfig.GetClusterNetworkConfig.GetVnetResourceGroup}}
+ROUTE_TABLE={{.ClusterConfig.GetClusterNetworkConfig.GetRouteTable}}
+PRIMARY_AVAILABILITY_SET={{.ClusterConfig.GetPrimaryAvailabilitySet}}
+PRIMARY_SCALE_SET={{.ClusterConfig.GetPrimaryScaleSet}}
+SERVICE_PRINCIPAL_CLIENT_ID={{.AuthConfig.GetServicePrincipalId}}
+NETWORK_PLUGIN={{getStringFromNetworkPluginType .GetNetworkConfig.GetNetworkPlugin}}
+NETWORK_POLICY="{{getStringFromNetworkPolicyType .GetNetworkConfig.GetNetworkPolicy}}"
+VNET_CNI_PLUGINS_URL={{.GetNetworkConfig.GetVnetCniPluginsUrl}}
+CNI_PLUGINS_URL={{.GetNetworkConfig.GetCniPluginsUrl}}
+CLOUDPROVIDER_BACKOFF=true
+CLOUDPROVIDER_BACKOFF_MODE=v2
+CLOUDPROVIDER_BACKOFF_RETRIES=6
+CLOUDPROVIDER_BACKOFF_EXPONENT=0
+CLOUDPROVIDER_BACKOFF_DURATION=5
+CLOUDPROVIDER_BACKOFF_JITTER=0
+CLOUDPROVIDER_RATELIMIT=true
+CLOUDPROVIDER_RATELIMIT_QPS=10
+CLOUDPROVIDER_RATELIMIT_QPS_WRITE=10
+CLOUDPROVIDER_RATELIMIT_BUCKET=100
+CLOUDPROVIDER_RATELIMIT_BUCKET_WRITE=100
+LOAD_BALANCER_DISABLE_OUTBOUND_SNAT={{.ClusterConfig.GetLoadBalancerConfig.GetDisableOutboundSnat}}
+USE_MANAGED_IDENTITY_EXTENSION={{.AuthConfig.GetUseManagedIdentityExtension}}
```

Comment:
Let's restore the defaulting to true as discussed in teams chat.

Reply:
Updated. https://github.com/Azure/karpenter-provider-azure/blob/02203647bd517608580e51a332abe77600b7c9af/pkg/providers/imagefamily/bootstrap/aksbootstrap.go#L226

Reply:
Technically this could be put on `staticNodeBootstrapVars` until we support it. But do you think otherwise?

Reply:
Agreed. Fixed. :)